### ISSUE TITLE
function-calc-no-unspaced-operator misinterprets - of negative number (without leading zero) as operator

### ISSUE THREAD

**Issue by woble** [stylelint/stylelint#1564]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Negative decimals below zero (without leading zero) inside a function, e.g. `calc(10em * -.5)`, don't work as expected when the following two rules are used in combination.

Obviously, one could divide by `-2` instead of multiplying by `-.5`, but that doesn't solve the underlying problem.

> Which rule, if any, is this issue related to?

`number-leading-zero: never`
`function-calc-no-unspaced-operator: true`

> What CSS is needed to reproduce this issue?

```css
.foo {
    left: calc(10em * -.5); /* function-calc-no-unspaced-operator warning */
    left: calc(10em * (-.5)); /* function-calc-no-unspaced-operator warning */

    /* For the sake of testing, knowing number-leading-zero is set to never */
    left: calc(10em * -0.5); /* number-leading-zero warning */
    left: calc(10em * (-0.5)); /* function-calc-no-unspaced-operator & number-leading-zero warnings */

    left: calc(10em * - .5); /* No warning, but undesired negative notation */
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
    "rules": {
        "number-leading-zero": "never",
        "function-calc-no-unspaced-operator": true
    }
}
```

> Which version of stylelint are you using?

`6.6.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`gulp-stylelint`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

See example.

**Comment by evilebottnawi** [comment]
@woble Thanks.

**Comment by jeddy3** [comment]
@woble Feel free to contribute a fix. You'll likely need to tweak `function-calc-no-unspaced-operator` operator detection so that `-.5` is parsed as a negative number.
